ksql-parser/src/main/java/io/confluent/ksql/parser/tree/SingleColumn.java

```diff
@@ -29,20 +29,20 @@
 @Immutable
 public class SingleColumn extends SelectItem {
 
-  private final ColumnName alias;
+  private final Optional<ColumnName> alias;
```

Comment:
Can you explain _why_ this is now optional, please?

Reply:
Previously, the parser would go and fill in aliases. However, the alias is sometimes based on the source name. Specifically, if the statement is a join, and a column with that name exists in both sources then the alias includes the source name. Since we don't have enough information to resolve this at parse time (without the meta store), we fill in aliases, and so this must be optional.

Reply:
I'm wondering if adding type safety around this makes sense - at some point, we expect all `SingleColumn` instances to have an alias. Maybe we have `ParsedColumn` containing the optional, and `SingleColumn` remains as is and is the output of rewriting? I'm not sure that's the right way to go, but I'm worried about using this to represent two different stages

Reply:
@agavra I thought about that a bit too. I don't think doing this actually adds much here. In either case you have to go from a reference to some common type (either SingleColumn w/ the optional, or a parent class) and check at runtime whether the alias has been filled in.

Reply:
So with your change does the parser now only set the alias here is the user actually supplied an alias in their statement?  That would be ideal from my point of view.

With regards to @agavra's comment... I think I agree with @agavra: we've been improving the code quality in the engine by leveraging type safety. This is making the engine much easier to reason about.   I think in this case the benefit is currently marginal, but adding it now also won't cost much and it could save pain later.

So I'd consider adding a `class ParsedColumn extends SelectItem` which has an optional alias, and then have the rewritter return a `SingleColumn` with a required alias.  We can then throw from downstream visitors if we see a `ParsedColumn`, (as they should have been re-written), and won't have got that's dereferencing an `Optional` alias all the time (i.e. it avoids either adding checks that `alias.isPresent` or disabling warnings that we're not checking).

Up to you.

Reply:
> So with your change does the parser now only set the alias here is the user actually supplied an alias in their statement?

Yes - the parser doesn't try to fill anything in. It just returns an AST that represents what was parsed.